_tools/libddwaf-updater/update.go

```diff
@@ -34,24 +35,35 @@ const (
 	goVersionUnsupported = "go1.23"
 )
 
+var (
+	forceFlag   *bool
+	versionFlag *string
+)
+
 func main() {
-	force := false
 
-	if len(os.Args) >= 2 {
-		force = os.Args[1] == "--force"
-	}
+	flag.Parse()
+
+	var (
+		release *github.RepositoryRelease
+		err     error
+		gh      = github.NewClient(nil)
+	)
 
-	gh := github.NewClient(nil)
+	if *versionFlag == "latest" {
+		release, _, err = gh.Repositories.GetLatestRelease(context.Background(), "DataDog", "libddwaf")
+	} else {
+		release, _, err = gh.Repositories.GetReleaseByTag(context.Background(), "DataDog", "libddwaf", *versionFlag)
+	}
```

Comment:
So you need not deref, which always looks dodgy w/o a `nil` check:

```suggestion
	if versionFlag == "latest" {
		release, _, err = gh.Repositories.GetLatestRelease(context.Background(), "DataDog", "libddwaf")
	} else {
		release, _, err = gh.Repositories.GetReleaseByTag(context.Background(), "DataDog", "libddwaf", versionFlag)
	}
```